src/rpc/mining.cpp

```diff
@@ -544,6 +544,7 @@ UniValue getblocktemplate(const JSONRPCRequest& request)
         pindexPrev = pindexPrevNew;
     }
     CBlock* pblock = &pblocktemplate->block; // pointer for convenience
+    assert(pblock);
```

Comment:
It is guaranteed to be true on the first call. But even if it wasn't, pblock would still never be NULL. It is always assigned to &pblocktemplate->block, which *might* be NULL with some platforms, but is not guaranteed to be, and if not would pass your assert yet still be wrong.